paging-runtime-composeui/src/commonMain/kotlin/app/cash/paging/LazyPagingItems.kt

```diff
@@ -0,0 +1,306 @@
+   */
+  fun peek(index: Int): T? {
+    return itemSnapshotList[index]
+  }
+
+  /**
+   * Retry any failed load requests that would result in a [LoadStateError] update to this
+   * [LazyPagingItems].
+   *
+   * Unlike [refresh], this does not invalidate [PagingSource], it only retries failed loads
+   * within the same generation of [PagingData].
+   *
+   * [LoadStateError] can be generated from two types of load requests:
+   *  * [PagingSource.load] returning [PagingSource.LoadResult.Error]
+   *  * [RemoteMediator.load] returning [RemoteMediator.MediatorResult.Error]
+   */
+  fun retry() {
+    pagingDataDiffer.retry()
+  }
+
+  /**
+   * Refresh the data presented by this [LazyPagingItems].
+   *
+   * [refresh] triggers the creation of a new [PagingData] with a new instance of [PagingSource]
+   * to represent an updated snapshot of the backing dataset. If a [RemoteMediator] is set,
+   * calling [refresh] will also trigger a call to [RemoteMediator.load] with [LoadType] [REFRESH]
+   * to allow [RemoteMediator] to check for updates to the dataset backing [PagingSource].
+   *
+   * Note: This API is intended for UI-driven refresh signals, such as swipe-to-refresh.
+   * Invalidation due repository-layer signals, such as DB-updates, should instead use
+   * [PagingSource.invalidate].
+   *
+   * @see PagingSource.invalidate
+   */
+  fun refresh() {
+    pagingDataDiffer.refresh()
+  }
+
+  /**
+   * A [CombinedLoadStates] object which represents the current loading state.
+   */
+  var loadState: CombinedLoadStates by mutableStateOf(
+    // See the mapped classes from androidx.paging -> app.cash.paging
+    // in multiplatform-paging/paging-common/README.md for more info
+    CombinedLoadStates(
+      refresh = InitialLoadStates.refresh,
+      prepend = InitialLoadStates.prepend,
+      append = InitialLoadStates.append,
+      source = InitialLoadStates
+    )
+  )
+    private set
+
+  internal suspend fun collectLoadState() {
+    pagingDataDiffer.loadStateFlow.filterNotNull().collect {
+      loadState = it
+    }
+  }
+
+  internal suspend fun collectPagingData() {
+    flow.collectLatest {
+      pagingDataDiffer.collectFrom(it)
+    }
+  }
+
+  private companion object {
+    init {
+      /**
+       * Implements the Logger interface from paging-common and injects it into the LOGGER
+       * global var stored within Pager.
+       *
+       * Checks for null LOGGER because other runtime entry points to paging can also
+       * inject a Logger
+       */
+      LOGGER = platformLogger()
+    }
+  }
+}
+
+// See the mapped classes from androidx.paging -> app.cash.paging
+// in multiplatform-paging/paging-common/README.md for more info
+private val IncompleteLoadState = LoadStateNotLoading(false)
+private val InitialLoadStates = LoadStates(
+  LoadStateLoading as LoadState,
+  IncompleteLoadState as LoadState,
+  IncompleteLoadState as LoadState
+)
+
```

Comment:
Just FYI - I also had to change this due to the workaround mentioned in [paging-common](https://github.com/cashapp/multiplatform-paging/tree/main/paging-common)